src/main/io/gps.c

```diff
@@ -521,50 +691,155 @@ static void ubloxSendPollMessage(uint8_t msg_id)
     tx_buffer.header.msg_class = CLASS_CFG;
     tx_buffer.header.msg_id = msg_id;
     tx_buffer.header.length = 0;
-    ubloxSendMessage((const uint8_t *) &tx_buffer, 6);
+    ubloxSendMessage((const uint8_t *) &tx_buffer, 6, false);
+}
+
+static void ubloxSendNAV5Message(uint8_t model) {
+    DEBUG_SET(DEBUG_GPS_UNIT_CONNECTION, 0, model);
+    ubxMessage_t tx_buffer;
+    if (gpsData.ubloxM9orAbove) {
```

Comment:
I strongly recommend splitting up functions/files based on feature.

This method mixed two entirely separate functions, which should be
- moved to a new, a separate file `gps_ublox.c`
- split across 3 separate methods
-  `ubloxSendNAV5Message` with `if/else`, a wrapper for the two methods
- `ubloxSendNAV5MessageM9orAbove`
- `ubloxSendNAV5MessageM8orBelow`

Literally all coding guidelines discourage long methods, please make a consious effort not to try to cram everything into a single function in a single file

Reply:
Significant refactoring is not within the scope of this PR, even though we agree entirely that it is needed.  Refactoring will be done only where it provides a fix for a functional problem, or arises naturally out of fixing a problem.